llvm/include/llvm/IR/Type.h

```diff
@@ -175,6 +187,9 @@ class Type {
     case HalfTyID:
     case BFloatTyID:
     case FP128TyID:
+    case Decimal32TyID:
+    case Decimal64TyID:
+    case Decimal128TyID:
```

Comment:
I don't think the decimal types should qualify as `isIEEELikeFPTy`--one of the requirements is that they don't have noncanonical values, and decimal FP types do have noncanonical values (like `x86_fp80`.

Reply:
That makes sense. 
IsFloatingType should then become:

`bool isFloatingPointTy() const {
    return isIEEELikeFPTy() || getTypeID() == X86_FP80TyID ||
           getTypeID() == PPC_FP128TyID || getTypeID() == Decimal32TyID ||
           getTypeID() == Decimal64TyID || getTypeID() == Decimal128TyID;
  }`

Reply:
May be we should have something like isDFPFLoatingType to check only for DFP?